server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -241,7 +243,11 @@ public int compareTo(DocIdToIndex o) {
         }
     }
 
-    private static FieldsVisitor createStoredFieldsVisitor(SearchContext context, Map<String, Set<String>> storedToRequestedFields) {
+    private static FieldsVisitor createStoredFieldsVisitor(
+        SearchContext context,
+        Map<String, Set<String>> storedToRequestedFields,
+        SourceLoader sourceLoader
+    ) {
```

Comment:
FieldsVisitor really needs to die as an API that we are forced to use anywhere...